plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/CreateQueryGroupRequest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.action;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+
+/**
+ * A request for create QueryGroup
+ * User input schema:
+ *  {
+ *    "name": "analytics",
+ *    "resiliency_mode": "enforced",
+ *    "resource_limits": {
+ *           "cpu" : 0.4,
+ *           "memory" : 0.2
+ *      }
+ *  }
+ *
+ * @opensearch.experimental
+ */
+public class CreateQueryGroupRequest extends ActionRequest implements Writeable.Reader<CreateQueryGroupRequest> {
```

Comment:
Do we need to implement `Writeable.Reader<CreateQueryGroupRequest>` interface?

Reply:
removed